applications/vanilla/controllers/class.discussionscontroller.php

```diff
@@ -88,7 +88,10 @@ public function index($Page = false) {
         $this->fireEvent('AfterPageCalculation');
 
         // Set canonical URL
-        $this->canonicalUrl(url(concatSep('/', 'discussions', pageNumber($Offset, $Limit, true, false)), true));
+        $canonicalUrl = $this->Data['isHomepage'] ?
```

Comment:
Protect against `isHomepage` key not existing.

Reply:
@tburry I'm pretty sure it always exists (ensured by `Gdn_Dispatcher` calling `->passData('isHomepage', $someValue)`)

I also don't think we have any intention of changing that due to the 100s of themes all rely on it being set all the time (and master views).